src/sections/image_resources_section.rs

```diff
@@ -0,0 +1,761 @@
+use std::collections::HashMap;
+use std::iter::Map;
+use std::ops::Range;
+
+use failure::{Error, Fail};
+
+use crate::sections::image_resources_section::UnitFloatStructure::Angle;
+use crate::sections::PsdCursor;
+
+const EXPECTED_RESOURCE_BLOCK_SIGNATURE: [u8; 4] = [56, 66, 73, 77];
+const RESOURCE_SLICES_INFO: i16 = 1050;
+
+struct ImageResourcesBlock {
+    resource_id: i16,
+    name: String,
+    data_range: Range<usize>,
+}
+
+#[derive(Debug)]
+pub struct ImageResourcesSection {
+    pub descriptors: Option<Vec<DescriptorStructure>>,
```

Comment:
What's the idea behind storing these are a vector, is there nothing to key them by in a map where they can be looked up by name?

Not rhetorical at all - just want to make sure I'm understanding the spec / data structures.

Thanks!

Reply:
No one PSD file in `fixtures/` does have more than one root descriptor. And this descriptor has an empty name field, I think it is not unique to use it as key.

Reply:
Got it